R/rt.R

```diff
@@ -19,3 +19,85 @@ rt_url <- function(base, ...) {
 #'
 #' @param l list
 compact <- function(l) Filter(Negate(is.null), l)
+
+
+#' Parse an RT response
+#'
+#' Parse an RT response
+#'
+#' @param resp_cont (character) The RT ticket response content
+#'
+
+parse_ticket <- function(resp_cont) {
+  #clean/split response
+  resp_split <- resp_cont %>%
+    stringr::str_replace_all("^RT.*Ok\n|\n\n$", "") %>% #remove headers/footers
+    stringr::str_split("\\n--\\n") %>%  #split if multiple tickets/etc displayed
+    unlist(recursive = FALSE)
+
+  resp_processed <- tryCatch(
+    lapply(resp_split,
+           function(.x){
+             tibble(fields = stringr::str_extract_all(.x, "\n[^: ]+:")[[1]] %>%
+                      str_replace_all("\\n|:", "") %>%
+                      trimws(),
+                    values = stringr::str_split(.x, "\n[^: ]+:")[[1]][-1] %>%
+                      trimws()) %>%
+               spread(fields, values)
+           }),
+    error = function(e) {NULL})
+
+  out <- dplyr::bind_rows(resp_processed)
+
+  if(nrow(out) == 0){
+    return(resp_split)
+    warning("The response could not be parsed into a tabular format")
+  } else {
+    return(out)
+  }
+}
+
+#' Get an RT response
+#'
+#' Get an RT response and format it into an S3 object
+#'
+#' @param url (character) The base URL that hosts RT for your organization
```

Comment:
I think this shouldn't be the base URL but the actual full URL

Reply:
good catch